### ISSUE TITLE
[SUGGESTION]: Notification should include icon of related element

### ISSUE THREAD

**Issue by YetiSpaghetti** [pokeclicker/pokeclicker#3904]
### Current Behavior

Notifications are generally just plain text (except for Shinies I believe, can't recall with certainty)

### Improved Behavior

Notifications should include a sprite icon of the associated element for the notification.

I could see this being used for
- [ ] New Pokemon Caught
- [ ] Egg hatched
- [ ] Egg ready to hatch (would be the matching egg icon if available)
- [ ] Shiny encountered (should be Shiny icon + Pokemon icon)
- [ ] Shiny caught (should be Shiny icon + Pokemon icon)
- [ ] Pokemon ready to hatch / hatched
- [ ] Berry ready / withering / dropped / etc

The icon should be specific to the associated element.

This principle could be applied to all notifications, even generic ones (such as Mining Energy Full) could use a generic mining icon.

### Reason for change

This change would result in better UX, faster visual recognition of what a notification means (rather that having to read the string). 

It also provides a small bit more joy to the player as well, seeing a sprite of a newly caught pokemon rather than just a plaintext string.

### Detailed Description

Ideally I think the icon would be at the left-side of the notification, to maintain a uniform aesthetic and rendering. Icons would be scaled to be uniform as well

example:
 
---
![image](https://user-images.githubusercontent.com/14989011/216516128-a9ac9534-4d53-4453-b6b3-2e541cead33c.png) Herracross caught!
---
![image](https://user-images.githubusercontent.com/14989011/216516331-27fff849-160d-4103-a716-2aae7f705aa8.png) Dratini is ready to hatch!
---
![image](https://user-images.githubusercontent.com/14989011/216516380-f5a179be-3855-4dc3-85d0-bc1af239f7e6.png) Oran Berry ready to harvest

**Comment by DraKay** [comment]
Wouldn't it better follow the style of item drop that already have a little icon onf a set size in the right of the text? Also you list lack dream orbs, the bullet points could be substituted by icons of the different types of dream orbs

**Comment by YetiSpaghetti** [comment]
@DraKay Yeah you're right, I didn't realize the notifications already have the icon on the right. It looks like the `Notification` interface accepts an `image` as an optional prop, so there is already a way to wire in the image for the notification (and use the default icon alignment). The question is whether the origination scope that calls the notifier has the image (or a way get it) that can be easily passed through. 

I'm not far enough in the game to know what dream orbs are, but sounds like they would also benefit from the icons as well :)

**Comment by YetiSpaghetti** [comment]
I've given this an attempt using the built-in `image` prop from the notifier, but the image sizing is way off (looks like the asset pngs have a ton of empty space around each pokemon).

<img width="357" alt="image" src="https://user-images.githubusercontent.com/14989011/216721973-68631007-4d4b-461f-823c-15f65c054463.png">

Any idea on how to fix this that doesn't involve regenerating 1000 pngs to remove the dead space? A hacky way would be to mess around with the `height` and `margin` css properties but that doesn't feel right.

**Comment by YetiSpaghetti** [comment]
Looks great for the Chesto Berry 😅 

Also note that Pidgeotto doesn't look as small as the others, so I suspect the assets themselves will have to be fixed due to inconsistent sizing.

<img width="357" alt="image" src="https://user-images.githubusercontent.com/14989011/216722544-7386e40f-f0a3-4b61-a079-b5f9b2d4d2f3.png">

**Comment by DraKay** [comment]
All pokemon images are 96x96 pidgiotto is just bigger than pidgey

**Comment by DraKay** [comment]
If you cut around them it will give problems in any other place

**Comment by YetiSpaghetti** [comment]
Yep it would require a big refactor to cut the pngs, definitely don't want to do that... It also makes sense for the battle screen, giving each mon a sense of scaled compared to each other.

How do we make the biggest and smallest pokemon sprites look scaled in this toast display... 🤔 

I suppose one option would be to duplicate the assets into a toast version, but that would also be bad for a variety of reasons 😓

**Comment by DraKay** [comment]
Actually is the format of Gen 5, as all sprites until that Gen are oficials

**Comment by CypherX** [comment]
@YetiSpaghetti - Just a heads up, this suggestion has not been approved (we have yet to triage it) so while you're welcome to implement it and open a PR there's no guarantee it will be accepted.

That said - the toast icon has a fixed height of 28px, maybe try adding the image directly into the message itself rather than as the icon.

**Comment by YetiSpaghetti** [comment]
Understood, mostly just using this to explore the codebase a bit, definitely know there's no guarantee of anything being approved. 😄 

What is the usual timeframe for a suggestion being triaged? And should I break this into smaller issues (one of each asset type) to make triaging easier?

> That said - the toast icon has a fixed height of 28px, maybe try adding the image directly into the message itself rather than as the icon.

It's less an issue of the fixed height, and more dealing with the fact that the pokemon sprite assets don't have consistent scaling (i.e. within the 96x96, a Pidgey is significantly smaller than a Charizard). 

The only way I can see providing consistent scale for the toast icon is to duplicate all the pokemon png assets and trim the empty space, that way each pokemon renders at the same scale in the toast, but that feels bad both from a maintainability and bundle size perspective.

BTW @CypherX love the profile pic. Believe in the me that believes in you 😎

**Comment by YetiSpaghetti** [comment]
I think I found a scaling solution that looks decent for both the largest and smallest size pokemon. 

Would love to hear what others think. In the below screenshot, Pidgeotto/Fearow are slightly smaller than the largest pokemon sprite, and Bulbasaur/Rattata are some of the smallest. For reference, the berries are the standard toast icon sizing. Everything is easy to perceive, and has consistent margins with the text, and the `pokemonIcons` do not change the standard height of toasts.

<img width="357" alt="image" src="https://user-images.githubusercontent.com/14989011/216733968-119914dd-c212-47af-9dea-3f76d9ef75a9.png">
